cirq/ops/linear_combinations.py

```diff
@@ -205,3 +207,54 @@ def extend(expansion: value.LinearDict[str],
             extended_expansion = extend(expansion, op.qubits, self.qubits)
             result += extended_expansion * coefficient
         return result
+
+
+class PauliSum:
+    """Represents operator defined by linear combination of PauliStrings.
+
+    Since PauliStrings store their own coefficients, this class
+    does not implement the LinearDict interface. Instead, you can
+    add and subtract terms and then iterate over the resulting
+    (simplified) expression.
+
+    Under the hood, this class is backed by a LinearDict with coefficient-less
+    PauliStrings as keys. PauliStrings are reconstructed on-the-fly during
+    iteration.
+
+    """
+
+    def __init__(self, linear_dict):
+        self._linear_dict = linear_dict
```

Comment:
What if the user passes in a linear dict containing Pauli strings that have non-unit coefficients?

Reply:
added validator